google-cloud-spanner/src/main/java/com/google/cloud/spanner/Statement.java

```diff
@@ -246,4 +248,56 @@ StringBuilder toString(StringBuilder b) {
     }
     return b;
   }
+
+  public static final class StatementFactory {
+    private final Dialect dialect;
+
+    StatementFactory(Dialect dialect) {
+      this.dialect = dialect;
+    }
+
+    public Statement of(String sql) {
+      return Statement.of(sql);
+    }
+
+    /**
+     * @param sql SQL statement with unnamed parameter denoted as ?
+     * @param values list of values which needs to replace ? in the sql
+     * @return Statement object
+     *     <p>This function accepts the SQL statement with unnamed parameters(?) and accepts the
+     *     list of objects to replace unnamed parameters. Primitive types are supported
+     *     <p>For Date column, following types are supported
+     *     <ul>
+     *       <li>java.util.Date
+     *       <li>LocalDate
+     *       <li>com.google.cloud.Date
+     *     </ul>
+     *     <p>For Timestamp column, following types are supported. All the dates should be in UTC
+     *     format. Incase if the timezone is not in UTC, spanner client will convert that to UTC
+     *     automatically
+     *     <ul>
+     *       <li>LocalDateTime
+     *       <li>OffsetDateTime
+     *       <li>ZonedDateTime
+     *     </ul>
+     *     <p>
```

Comment:
It is customary to add this kind of description at the start of the Javadoc, before the parameters and return type tags, so this block should be moved up.

```suggestion
     *     <p>This function accepts a SQL statement with unnamed parameters (?), and accepts a
     *     positional list of objects as the values of the unnamed parameters. Primitive types are supported.
     *     <p>For Date values, the following types are supported:
     *     <ul>
     *       <li>java.util.Date
     *       <li>java.time.LocalDate
     *       <li>com.google.cloud.Date
     *     </ul>
     *     <p>For Timestamp values, the following types are supported:
     *     <ul>
     *       <li>LocalDateTime
     *       <li>OffsetDateTime
     *       <li>ZonedDateTime
     *     </ul>
     *     <p>
```

We also need a comment on how `LocalDateTime` is interpreted. The current implementation interprets it in UTC, but I'm not sure that is the most logical. `LocalDateTime` is intended to be interpreted using the local timezone, so we should consider that here as well.

Reply:
Done